packages/studio/src/components/PropInput.tsx

```diff
@@ -61,17 +58,16 @@ export default function PropInput({
   switch (type) {
     case PropValueType.number:
       return (
-        <input
-          type="number"
-          onChange={handleChangeEvent}
-          className={inputBoxCssClasses}
-          value={(propValue ?? "") as number | string}
+        <PopulatePropInput
```

Comment:
I think it might make more sense to have a separate NumberPropInput component? Instead of using the same component for strings and numbers

Reply:
I could! I was debating, but the code between the two would be pretty similar.

Reply:
Ideally the component for string props wouldn't have any code only used for number props and vice versa. You could still share code between the two of them via a shared component or hook. Alternatively you could turn the logic that is different between the two use cases as props instead of having switching logic within a component. At that point you'd probably just want two separate components though